assets/js/data/cart/actions.ts

```diff
@@ -484,22 +485,24 @@ export const updateCustomerData =
 				data: customerData,
 				cache: 'no-store',
 			} );
-
-			dispatch.receiveCartContents( response );
+			if ( editing ) {
+				dispatch.receiveCartContents( response );
+			} else {
+				dispatch.receiveCart( response );
+			}
+			dispatch.updatingCustomerData( false );
 		} catch ( error ) {
 			dispatch.receiveError( error );
+			dispatch.updatingCustomerData( false );
 
 			// If updated cart state was returned, also update that.
 			if ( error.data?.cart ) {
 				dispatch.receiveCart( error.data.cart );
 			}
 
-			// rethrow error.
-			throw error;
-		} finally {
-			dispatch.updatingCustomerData( false );
+			return Promise.reject( error );
 		}
-		return true;
+		return Promise.resolve( true );
```

Comment:
I added promises here so we can use them when consuming `updateCustomerData`. Is this pattern ok @senadir ?

```
dispatch( CART_STORE_KEY ).updateCustomerData( xxx ).then( xxx );
```

Reply:
I'm not sure, in my understanding you should yield here but this is not a generator, I haven't seen this pattern before, okay to keep unless it causes an issue later.